packages/web-components/fast-foundation/src/accordion-item/accordion-item.template.ts

```diff
@@ -30,14 +35,13 @@ export function accordionItemTemplate(
                         <slot name="heading"></slot>
                     </span>
                 </button>
-                ${startSlotTemplate(options)}
                 ${endSlotTemplate(options)}
-                <span class="icon" part="icon" aria-hidden="true">
+                <span class="expand-collapse-icon" part="expand-collapse-icon" aria-hidden="true">
                     <slot name="expanded-icon">
-                        ${options.expandedIcon || ""}
+                        ${options.expandedIcon || expandedIcon}
```

Comment:
I'm not sure these will be tree shakeable in this instance. Do we have an idea of how this affects the overall size of the package?

Reply:
This won't be tree-shakeable and I think we should handle this differently. Instead, just provide the icon as an independent export not baked into the template. Then the CLI, when it generates the default setup for the component will import that icon and pass it through the options. This will allow that icon to be used by default but overridden and tree-shaken by folks building a system that wants a different icon. /cc @janechu

Reply:
I minimized them and they are generally very simple shapes. They're small, but I didn't check exactly how many kb. We already inline the search clear icon. I suspect with some of the template optimizations the changes net out.

Thinking another way, we add code and elements often without questioning the size. From a functional standpoint these minor additions really reduce the barrier to getting up and running. I considered some shared location but couldn't figure out a way those would tree-shake either.

Reply:
> Thinking another way, we add code and elements often without questioning the size.

We have in the past, and I think that's been a problem we need to watch out for closely. One of our values is enabling great performance and we should prevent an approach that impacts that where we can. Developer experience is important, but user experience is *key* for these scenarios. I don't think that requiring a developer to slot these in or pass them via options hampers devex enough to justify littering the bundle with baked SVG's. If we want to provide them from Foundation, I'm open to that - but this one is a hard no from me simply because it assumes the trade-off is worth it, and I just don't think it is here.

Reply:
@EisenbergEffect, so in this model the registration would still need to provide the value to the `options` but at least if someone uses the CLI they will get that and be able to override it? I was looking for a solution where the registration didn't even need the value because I think it's still error-prone and you end up with a broken component that's confusing. Like a checkbox with no check. It makes it a hard requirement to use the CLI rather than hand-roll a component.

I considered a wrapper for the `<svg>` boilerplate as well. We really only need the `path`, and there are only a few of them we could put in a common location. Mainly chevrons. It would be a very small addition and easy to replace still.

Reply:
IMO the concept of this being error-prone is a personal consideration or concern rather than something objective. If someone wants to go an old school route of injecting via `:before`, `:after`, or other routes I think that needs to be acceptable. I also think we're assuming an audience that isn't broad. For some, the CLI will provide a great experience and affordance that wires a lot of this up for free. Foundation should not hoist the implementation details of composing these though, that is a lot of opinion and assumption for how someone may want to use it and their experience.

Further, good examples and documentation can help create a great experience while allowing the code to remain unopinionated.

> We value both composition and performance, so while we provide all the building blocks to bring your component together, we realize that you may want to customize beyond what we consider to be "common". Given this, the Foo component supports slots for icons which can be configured by leveraging the default FooExpandedIcon which is exported from the package. If you'd like to use your own icon, you can pass that via the options as part of configuring your component. Further, if you have unique one-off scenarios, you can pass the a default to the option configuration and override it by slotting a custom svg or icon when necessary.

There are different paths for consuming and creating and there are different skill levels of folks for sure. Ideally our documentation can be improved as part of this to help make it easier for folks wanting to start w/ just Foundation. The CLI essentially wires everything up with our ideals - we don't need to hoist those ideals into Foundation IMO.